pkg/core/resources/manager/cache.go

```diff
@@ -85,6 +85,9 @@ func (c *cachedManager) Get(ctx context.Context, res model.Resource, fs ...store
 
 func (c *cachedManager) List(ctx context.Context, list model.ResourceList, fs ...store.ListOptionsFunc) error {
 	opts := store.NewListOptions(fs...)
+	if !opts.IsCacheable() {
+		return fmt.Errorf("filter functions are not allowed for cached store")
```

Comment:
Why do we return an error here? It means you can't use `ListByFilterFunc` at all (even if you're okay to get non-cached result)

Reply:
The point is to avoid caching filtered data. In case you provide filters you might cache filtered data or retrieve data that wasn't filtered.

Reply:
yeah, but why not just delegate the call to `c.delegate` without caching?

Reply:
To avoid unforseen bad performance. If you using the cache you expect things to be cached so we explicitly fail if it's not cached.

Reply:
Should we just delete `ListByFilterFunc`?

Reply:
`List` interface is implemented in many different implementations and would require a lot of changes.

Reply:
But it's not part of the interface, I see that we use `ListByFilterFunc` only twice (and one usage you already removed in this PR). Can we rewrite the second usage of `ListByFilterFunc` and remove this option?

Reply:
But what about someone is going to create a new method that calls `List` and provides filter function? It might introduce a new error.

Reply:
There are 2 cases where the store api is used:

1. in the api-server (no cache and use the filterFunc and pagination api)
2. in the XDS server (leverages cache and should use filterFun)

Unfortunately the Store interface was abused and I think just rejecting unsupported filters is good enough (especially if we have good testing in place)

Reply:
Store.List is used in the api-server where you do need to be able to pass the option. Here for example is a genuine usage of it: https://github.com/kumahq/kuma/blob/master/pkg/api-server/filtering.go#L55-L75

Reply:
Yes, but it's used only here https://github.com/kumahq/kuma/blob/master/pkg/api-server/dataplane_overview_endpoints.go#L131. And you can rewrite it as:
```go
dataplanes := mesh.DataplaneResourceList{}
if err := r.resManager.List(ctx, &dataplanes, store.ListByMesh(meshName), store.ListByPage(p.size, p.offset), store.ListByNamePrefix(namePrefix)); err != nil {
	return mesh.DataplaneOverviewResourceList{}, err
}

filtered := mesh.DataplaneResourceList{}
for _, dp := range dataplanes.Items {
	if filter(dp) {
		filtered.AddItem(dp)
	}
}
```

Reply:
No because the filter makes the pagination works correctly (the total and page count).

Reply:
Also filters are shared between ResourceStore and ReadOnlyResourceStore, and it might be useful to pass them to resourcestore because we can do filtering in e.g. Postgres